tests/test_stunnel.py

```diff
@@ -0,0 +1,179 @@
+from pytest_container.container import ContainerData
+from pytest_container.pod import Pod
+from pytest_container.pod import PodData
+
+from bci_tester.data import STUNNEL_CONTAINER
+
+_stunnel_ctr, _marks = container_and_marks_from_pytest_param(STUNNEL_CONTAINER)
+_stunnel_kwargs = _stunnel_ctr.__dict__
+for key in [
+    "custom_entry_point",
+    "extra_environment_variables",
+    "containerfile",
+    "forwarded_ports",
+]:
+    _stunnel_kwargs.pop(key)
+
+_cert_dir = Path(__file__).parent / "files" / "stunnel"
+_SERVER_PEM = str(_cert_dir / "server.pem")
+
+_stunnel_kwargs["volume_mounts"].extend(
+    [
+        BindMount(
+            container_path="/etc/stunnel/stunnel.pem",
+            host_path=_SERVER_PEM,
+        ),
+        BindMount(
+            container_path="/etc/stunnel/stunnel.key",
+            host_path=str(_cert_dir / "server.key"),
+        ),
+    ]
+)
+
+_STUNNEL_PYTHON_HTTP_TUNNEL_CTR = DerivedContainer(
+    **_stunnel_kwargs,
+    extra_environment_variables={
+        "STUNNEL_DEBUG": "info",
+        "STUNNEL_SERVICE_NAME": "python",
+        "STUNNEL_ACCEPT": "0.0.0.0:8443",
+        "STUNNEL_CONNECT": "0.0.0.0:8000",
+    },
+)
+
+_PYTHON_WEB_SERVER = DerivedContainer(
+    base="registry.suse.com/bci/python",
+    containerfile="""
+RUN echo "hello world" > index.html
+ENTRYPOINT ["python3", "-m", "http.server"]
+""",
+)
+
+
+_STUNNEL_POD = pytest.param(
+    Pod(
+        containers=[_PYTHON_WEB_SERVER, _STUNNEL_PYTHON_HTTP_TUNNEL_CTR],
+        forwarded_ports=[PortForwarding(container_port=8443)],
+    ),
+    marks=_marks,
+)
+
+
+@pytest.mark.parametrize("pod", [_STUNNEL_POD], indirect=True)
+def test_stunnel_http_proxy(pod: PodData):
+    """Smoke test for stunnel to wrap a python http server in a TLS connection
+    using the generated certificate. The python web server is running in its own
+    container and stunnel in another one both in a single pod that only exposes
+    the TLS endpoint.
+
+    """
+
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(5), wait=tenacity.wait_exponential()
+    )
+    def get_request_to_stunnel(route: str = "") -> requests.Response:
+        resp = requests.get(
+            f"https://127.0.0.1:{pod.forwarded_ports[0].host_port}/{route}",
+            verify=_SERVER_PEM,
+            timeout=5,
+        )
+        resp.raise_for_status()
+        return resp
+
+    assert "hello world" in get_request_to_stunnel().text
+
+
+_CTR_PORT = 8000
+
+
+def _create_http_forward_ctr(log_level: str = "info") -> ParameterSet:
+    return pytest.param(
+        DerivedContainer(
+            **_stunnel_kwargs,
+            forwarded_ports=[PortForwarding(container_port=_CTR_PORT)],
+            extra_environment_variables={
+                "STUNNEL_DEBUG": log_level,
+                "STUNNEL_SERVICE_NAME": "wikipedia_dot_org",
+                "STUNNEL_ACCEPT": f"0.0.0.0:{_CTR_PORT}",
+                "STUNNEL_CONNECT": "wikipedia.org:80",
```

Comment:
I would recommend to use some of our own resources, e.g. conncheck.opensuse.org or something similar and avoid using wikipedia for reasons other than it's actual purpose. Even when it's not directly used now, it could be used by accident.

Otherwise LGTM.
```suggestion
                "STUNNEL_CONNECT": "conncheck.opensuse.org:80",
```